control_panel_api/serializers.py

```diff
@@ -94,11 +73,76 @@ def update(self, instance, validated_data):
         return super().update(instance, validated_data)
 
 
+class SimpleAppSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = App
+        fields = (
+            'id',
+            'url',
+            'name',
+            'slug',
+            'repo_url',
+            'iam_role_name',
+            'created_by',
+        )
+
+
+class SimpleS3BucketSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = S3Bucket
+        fields = ('id', 'url', 'name', 'arn', 'created_by')
+
+
+class AppAppS3BucketSerializer(serializers.ModelSerializer):
+    """Used from within with AppSerializer to not expose app"""
+    s3bucket = SimpleS3BucketSerializer()
+
+    class Meta:
+        model = AppS3Bucket
+        fields = ('id', 'url', 's3bucket', 'access_level')
+
+
+class S3BucketAppS3BucketSerializer(serializers.ModelSerializer):
+    """Used from within with S3BucketSerializer to not expose s3bucket"""
+    app = SimpleAppSerializer()
+
+    class Meta:
+        model = AppS3Bucket
+        fields = ('id', 'url', 'app', 'access_level')
+
+
+class AppSerializer(serializers.ModelSerializer):
+
+    apps3buckets = AppAppS3BucketSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = App
+        fields = (
+            'id',
+            'url',
+            'name',
+            'slug',
+            'repo_url',
+            'iam_role_name',
+            'created_by',
+            'apps3buckets',
+            'userapps',
```

Comment:
At some point we'll want the `userapps` to behave similarly for consistency as well.

Reply:
Yes if we need the extra data for @axemonkey we can do that later.

Reply:
I think for now this is what he needs, let's see how he uses some of the other endpoints if he needs extra data, then we can add this approach or refactor to a shared implementation.